### PR TITLE
Fix sanity check done on configured filesystem directories when running Alertmanager in microservices mode (#2947)

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
Cherry-picking the fix done in #2947.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ### Grafana Mimir
 
+* [BUGFIX] Fix sanity check done on configured filesystem directories when running Alertmanager in microservices mode. #2947
```
Comment (pstibrany) [inline CHANGELOG.md:7]: Changelog on the release branch should not mention `main / unreleased`.

Comment (pracucci) [reply CHANGELOG.md:7]: But it does. I guess we'll fix it before publishing the stable version.

Comment (pstibrany) [reply CHANGELOG.md:7]: > But it does.

Yep. 😂 (It has nothing to do with this PR, I should have been more careful reviewing the other PR that introduced the change.)

**Review by jhesketh — APPROVED** [review]
Seems reasonable as far as a backport goes, but take with a grain of salt as I'm still learning the internals.

**Pull request merged**